fwoa-tools/src/migrationUtils.test.ts

```diff
@@ -0,0 +1,156 @@
+      );
+    });
+  });
+
+  describe('getFhirClientSMART', () => {
+    test('throws error when environment variable is missing', async () => {
+      await expect(getFhirClientSMART()).rejects.toThrowError(
+        'SMART_AUTH_USERNAME environment variable is not defined'
+      );
+      process.env.SMART_AUTH_USERNAME = 'fake-smart-user-name';
+      await expect(getFhirClientSMART()).rejects.toThrowError(
+        'SMART_AUTH_PASSWORD environment variable is not defined'
+      );
+      process.env.SMART_AUTH_PASSWORD = 'fake-password';
+      await expect(getFhirClientSMART()).rejects.toThrowError(
+        'SMART_CLIENT_ID environment variable is not defined'
+      );
+      process.env.SMART_CLIENT_ID = 'fake-client-id';
+      await expect(getFhirClientSMART()).rejects.toThrowError(
+        'SMART_CLIENT_SECRET environment variable is not defined'
+      );
+      process.env.SMART_CLIENT_SECRET = 'fake-client-secret';
+      await expect(getFhirClientSMART()).rejects.toThrowError(
+        'SMART_OAUTH2_API_ENDPOINT environment variable is not defined'
+      );
+      process.env.SMART_OAUTH2_API_ENDPOINT = 'fake-oauth-endpoint';
+      await expect(getFhirClientSMART()).rejects.toThrowError(
+        'SMART_SERVICE_URL environment variable is not defined'
+      );
+      process.env.SMART_SERVICE_URL = 'fake-service-url';
+      await expect(getFhirClientSMART()).rejects.toThrowError(
+        'SMART_API_KEY environment variable is not defined'
+      );
+    });
+  });
+
+  describe('checkConfiguration', () => {
+    test('throws error when environment variable is missing', async () => {
+      const logs: WriteStream = createWriteStream(`unit-test.log`, {
+        flags: 'a'
+      });
+      await expect(checkConfiguration(logs)).rejects.toThrow(
+        'Environment variables EXPORT_BUCKET_NAME, ' +
+          'BINARY_BUCKET_NAME, GLUE_JOB_NAME, DATASTORE_ID, DATASTORE_ENDPOINT, DATA_ACCESS_ROLE_ARN, ' +
+          'IMPORT_KMS_KEY_ARN, IMPORT_OUTPUT_S3_BUCKET_NAME, HEALTHLAKE_CLIENT_TOKEN, EXPORT_BUCKET_URI, ' +
+          'IMPORT_OUTPUT_S3_URI not defined.'
+      );
+    });
+
+    test('should check s3 and healthlake ', async () => {
+      process.env.EXPORT_BUCKET_NAME = 'fake-export-bucket-name';
+      process.env.BINARY_BUCKET_NAME = 'fake-binary-bucket-name';
+      process.env.API_AWS_REGION = 'us-east-1';
+      process.env.GLUE_JOB_NAME = 'fake-job-name';
+      process.env.DATASTORE_ID = 'fake-ds-id';
+      process.env.DATASTORE_ENDPOINT = 'fake-endpoint';
+      process.env.DATA_ACCESS_ROLE_ARN = 'fake-role-arn';
+      process.env.IMPORT_KMS_KEY_ARN = 'fake-kms-arn';
+      process.env.IMPORT_OUTPUT_S3_BUCKET_NAME = 'fake-import-output-bucket-name';
+      process.env.HEALTHLAKE_CLIENT_TOKEN = 'fake-client-id';
+      process.env.EXPORT_BUCKET_URI = 'fake-export-uri';
+      process.env.IMPORT_OUTPUT_S3_URI = 'fake-import-uri';
+      const logs: WriteStream = createWriteStream(`unit-test.log`, {
```

Comment:
Are these files added to the gitignore btw?

Reply:
*.log are already in .gitignore